### ISSUE TITLE
Support for blank nodes to add text, maths, etc.

### ISSUE THREAD

**Issue by RazinShaikh** [zxcalc/zxlive#268]
This should be solely decorative and should not affect any functionality.

**Comment by RazinShaikh** [comment]
Should we also support edges between blank nodes? For example, the user might want to highlight a particular part of the diagram by drawing a box around it.

**Comment by RazinShaikh** [comment]
And how should these things work in the proof mode? Can we have different decorations for each step of the proof? There are too many possibilities, we should discuss the desiderata.

**Comment by jvdwetering** [comment]
Yeah it might be good discussing this in a bit more detail in person. I think having different decorations for every step makes a lot of sense. I imagine if you start with some decoration and do a proof step, that this decoration should be copied to the next step. But then there you can modify the decoration while keeping it the same proof step.
